lib/src/cli.dart

```diff
@@ -74,28 +75,40 @@ ArgParser createArgsParser() {
       help: "print the version of grinder");
   parser.addFlag('help', abbr: 'h', negatable: false,
       help: "show targets but don't build");
-  parser.addFlag('deps', abbr: 'd', negatable: false,
+  // TODO: Deprecated; remove post 0.7.0.
+  parser.addFlag('deps', abbr: 'd', negatable: false, hide: true,
       help: "display the dependencies of targets");
   return parser;
 }
 
-void printUsage(ArgParser parser, Grinder grinder) {
+void printUsageAndDeps(ArgParser parser, Grinder grinder) {
   print('usage: dart ${currentScript()} <options> target1 target2 ...');
   print('');
   print('valid options:');
   print(parser.usage.replaceAll('\n\n', '\n'));
 
-  if (!grinder.tasks.isEmpty) {
+  if (grinder.tasks.isEmpty) {
+    print('');
+    print('no current grinder targets');
+  } else {
+    // calculate the dependencies
+    grinder.start([], dontRun: true);
+
+    print('');
+    print('targets:');
     print('');
-    print('valid targets:');
 
     List<GrinderTask> tasks = grinder.tasks.toList();
-    tasks.forEach((t) {
-      var buffer = new StringBuffer()..write('  $t');
-      if (grinder.defaultTask == t) buffer.write(' (default)');
-      if (t.description != null) buffer.write(' ${t.description}');
-      print(buffer.toString());
-    });
+    print(tasks.map((task) {
+      bool isDefault = grinder.defaultTask == task;
+      Iterable<GrinderTask> deps = grinder.getImmediateDependencies(task);
+
+      String str = '${task}\n';
+      if (isDefault) str += '  @DefaultTask\n';
```

Comment:
The `@` callouts were mostly for to make it visually interesting; I changed it to `(defaults)` and `depends on` instead. I definitely want to make it pretty, as well as compact. I'll land as is and we can experiment with some other styles if necessary.